truffle/src/main/java/org/jruby/truffle/RubyFiletypeDetector.java

```diff
@@ -0,0 +1,46 @@
+package org.jruby.truffle;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.spi.FileTypeDetector;
+import java.util.Optional;
+
+public final class RubyFiletypeDetector extends FileTypeDetector {
+
+    private final static String[] KNOWN_RUBY_FILES = new String[]{ "Gemfile", "Rakefile", "Mavenfile" };
+    private final static String[] KNOWN_RUBY_SUFFIXES = new String[]{ ".rb", ".rake", ".gemspec" };
+    private final static String RUBY_MIME = "application/x-ruby";
+
+    @Override
+    public String probeContentType(Path path) throws IOException {
+        final Path fileNamePath = path.getFileName();
+        if (fileNamePath == null) {
+            return null;
+        }
+
+        final String fileName = fileNamePath.toString();
+
+        for (String candidate : KNOWN_RUBY_SUFFIXES) {
+            if (fileName.toLowerCase().endsWith(candidate)) {
+                return RUBY_MIME;
+            }
+        }
+
+        for (String candidate : KNOWN_RUBY_FILES) {
+            if (fileName.equals(candidate)) {
+                return RUBY_MIME;
+            }
+        }
+
+        final Optional<String> firstLine = Files.lines(path).findFirst();
+        if (firstLine.isPresent() && firstLine.get().equals("#!/usr/bin/env ruby")) {
+            return RUBY_MIME;
+        }
+
```

Comment:
Nice, but maybe too strict. What about two spaces in between "env" and "ruby". What if somebody uses just "#!/usr/bin/ruby"...